### DISCUSSION TITLE
jsonata to filter data using variable

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana-infinity-datasource#329] [Q&A] [answered]
Greetings @yesoreyeram 

Could you please help with how one can use jsonata to filter the Covid worldwide data by county?

COVid data:
https://api.covid19api.com/summary

Countries (to use for countries dashboard variable)
https://raw.githubusercontent.com/google/dspl/master/samples/google/canonical/countries.csv

or 

https://dzone.com/articles/list-of-all-countries-in-json as inline json

How could one then use this countries variable $country in a jsonata filter in infinity?

Thanks!

**Comment by yesoreyeram** [comment]
I ended up with the following UQL as shown in [here](https://grafana-infinity-datasource.herokuapp.com/d/F0JTk_j7z/jsonata-filter?orgId=1)

```kusto
parse-json
| scope "Countries"
| jsonata "*[Country='$country']"
| project kv()
```

There is also a related bug in UQL https://github.com/yesoreyeram/uql/issues/5 where filtering data produce single object instead array.

**Reply by yesoreyeram** [reply]
Hi @yosiasz - If you want to filter on multiple results, you can use similar syntax

```kusto
parse-json
| scope "Countries"
| jsonata "*[Country in [${countries:singlequote}]]"
```

**Reply by yosiasz** [reply]
Yep, I remember that bug 😄 no worries, no rush. I have found to get around that by using transformation 

![image](https://user-images.githubusercontent.com/900951/174190466-26060505-a955-4c56-8fa8-892fdb25b0b9.png)

Thanks so much!

**Reply by yesoreyeram** [reply]
I have updated the demo dashboard with more examples. But glad to know you figured out already.

**Reply by yesoreyeram** [reply, marked as answer]
BTW, I found a workaround for the issue that UQL returning single element. 

Use `[]` at the end of the expression. For example

use `scope "Countries" | jsonata "*[Country='$country'][]` instead of `scope "Countries" | jsonata "*[Country='$country']`

**Comment by shreya1110-dev** [comment]
Greetings @yesoreyeram,

I have a use case where I need to filter variable A based on variable B, such that if I select one or more options in B, they should not be included in variable A.

eg: I need to do something like 
Available options for Variable A : 1,2,3,4,5,6,7
Selected options for Variable A: 2,3,4
Available options for Variable B: 1,5,6,7,.. (can have other values)

```
parse-json
| scope "Countries"
| jsonata "*[Country not in [${countries:singlequote}]]"
```
I tried to modify this query to use ```!in``` or ```not in```, however I am not able to achieve my result. Is there anything I'm missing out?

**Reply by yesoreyeram** [reply]
Hiya.. It is hard fix your query unless we know what is your data looks like. BTW, nested variables example is available [here](https://play.grafana.org/d/c8416e7a-7cf3-48a1-b714-4a31ce7d0f5c/nested-variables-chained-variables?orgId=1).
